usefulUtilities/userLocation/userLocation.js

```diff
@@ -0,0 +1,208 @@
+
+
+(function () {
+    var request = Script.require("./modules/request.js").request;
+
+    var _this;
+    var REFRESH_LOCATION_INTERVAL_MS = 30000;
+    var USER_API_BASE_URL = Account.metaverseServerURL + "/api/v1/users?filter=connections&search=";
+    var TEXT_LOCAL_ENTITY_PROPS = {
+        "type": "Text",
+        // eslint-disable-next-line no-magic-numbers
+        "localPosition": [0, 0, 0.05],
+        // eslint-disable-next-line no-magic-numbers
+        "localRotation": Quat.fromVec3Degrees([0, 0, 0]),
+        "lineHeight": 0.08,
+        // eslint-disable-next-line no-magic-numbers
+        "dimensions": [1.5, 0.2, 0.05],
+        "topMargin": 0,
+        "rightMargin": 0,
+        "bottomMargin": 0,
+        "leftMargin": 0,
+        "grab": {
+            "grabbable": false,
+            "equippableLeftRotation": {
+                "x": -0.0000152587890625,
+                "y": -0.0000152587890625,
+                "z": -0.0000152587890625,
+                "w": 1
+            },
+            "equippableRightRotation": {
+                "x": -0.0000152587890625,
+                "y": -0.0000152587890625,
+                "z": -0.0000152587890625,
+                "w": 1
+            }
+        },
+        "canCastShadow": false,
+        "script": Script.resolvePath("userLocationTextLocalEntity.js?6")
+    };
+
+    var UserLocationBanner = function() {
+        _this = this;
+    };
+
+    UserLocationBanner.prototype = {
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            _this.targetUsername = false;
+            _this.targetDisplayName = false;
+            _this.refreshLocationInterval = false;
+            _this.textLocalEntity = false;
+            _this.canTeleportToTargetUser = false;
+    
+            var properties = Entities.getEntityProperties(entityID, ["userData"]);
+            var userData;
+    
+            try {
+                userData = JSON.parse(properties.userData);
+            } catch (e) {
+                console.error("Error parsing userData: ", e);
+            }
+    
+            if (userData) {
+                if (userData.targetUsername && userData.targetUsername.length > 0) {
+                    _this.targetUsername = userData.targetUsername;
+                } else {
+                    console.log("Please specify `targetUsername` inside this entity's `userData`!");
+                    return;
+                }
+    
+                if (userData.targetDisplayName && userData.targetDisplayName.length > 0) {
+                    _this.targetDisplayName = userData.targetDisplayName;
+                } else {
+                    console.log("Please specify `targetDisplayName` inside this entity's `userData`!");
+                    return;
+                }
+            } else {
+                console.log("Please specify this entity's `userData`! See README.md for instructions.");
```

Comment:
I wish we could get feedback like this not in the console.  Maybe we can use notifications for these, or something else that doesn't depend on the console to see why there was an error.

Reply:
i agree...notifications seem like the right move, if our notification system were better. i'll think about this for future projects. great idea